libzfs.pyx

```diff
@@ -1784,12 +1784,6 @@ cdef class ZFSVdev(object):
         if self.type == zfs.VDEV_TYPE_FILE:
             raise ZFSException(Error.NOTSUP, "Can replace disks only")
 
-        if not self.parent:
-            raise ZFSException(Error.NOTSUP, "Cannot replace a top-level vdev")
-
-        if self.parent.type not in (zfs.VDEV_TYPE_MIRROR, 'raidz1', 'raidz2', 'raidz3'):
-            raise ZFSException(Error.NOTSUP, "Can replace disks in mirror and raidz* vdevs only")
```

Comment:
We prob shouldnt remove that last condition just change it to `self.parent and ...`
Otherwise it may try to let you replace disks within a cache/spare/log? Or is that not the case

Reply:
Well, nevermind, it shouldnt matter

Reply:
`zpool_vdev_attach` handles the cache case and i think the spare one as well